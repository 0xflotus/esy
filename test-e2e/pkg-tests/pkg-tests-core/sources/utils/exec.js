/* @flow */

const cp = require('child_process');

exports.execFile = function(
  path: string,
  args: Array<string>,
  options: Object,
): Promise<{|stdout: Buffer, stderr: Buffer|}> {
  return new Promise((resolve, reject) => {
    cp.execFile(path, args, options, (error, stdout, stderr) => {
      if (error) {
        reject(`
          ${error}
          STDOUT: ${stdout}
          STDERR: ${stderr}
        `);
      } else {
        resolve({stdout, stderr});
      }
    });
  });
};
