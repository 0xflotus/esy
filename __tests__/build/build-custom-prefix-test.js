/**
 * @flow
 */

import * as path from 'path';
import {defineTestCaseWithShell} from '../utils';

defineTestCaseWithShell(
  path.join(__dirname, 'fixtures', 'custom-prefix'),
  `
    unset ESY__PREFIX
    run esy build
    assertStdout "esy x custom-prefix" "custom-prefix"
  `,
  {snapshotExecutionTrace: true},
);
