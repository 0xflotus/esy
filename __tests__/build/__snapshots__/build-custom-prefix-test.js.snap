// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build custom-prefix 1`] = `
"RUNNING: esy config ls
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/esy
RUNNING: esy build
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/project/store
RUNNING: esy x custom-prefix
esy:bin using <sandbox>/.esyrc
esy:bin prefix: <testRoot>/project/store"
`;
